### PR TITLE
Various core types

### PR DESCRIPTION
**Pull request by amulware**
What it says on the tin.

### CODE REVIEW

File: src/Core/Box.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+namespace Bearded.Utilities
+{
+    /// <summary>
+    /// Static Box class to help with box creation.
+    /// </summary>
+    public static class Box
+    {
+        /// <summary>
+        /// Gets a new box witha  given value.
```
Comment (tomrijnbeek) [inline src/Core/Box.cs]: typo

File: src/Core/Box.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        /// </summary>
+        /// <param name="value">The value of the box.</param>
+        public static Box<T> This<T>(T value)
```
Comment (tomrijnbeek) [inline src/Core/Box.cs]: is this supposed to be an extension method?

Comment (amulware) [inline src/Core/Box.cs]: Nope, intended usage is:

`var t = Box.This(DateTime.Now);`

Do you think I should turn it into an extension method (would be valid on all value types).
Or should I rename it to `From` or something else instead?

Comment (tomrijnbeek) [inline src/Core/Box.cs]: I see. Is fine then, didn't understand :)

**Pull request merged**